File: src/main/java/cz/wa/secretagent/io/campaign/model/SimpleModelCreator.java
package cz.wa.secretagent.io.campaign.model;

import java.util.List;
import java.util.Map;

import cz.wa.secretagent.io.tiles.singleproperties.ModelProperties;
import cz.wa.secretagent.view.TileId;
import cz.wa.secretagent.worldinfo.graphics.ModelInfo;
import secretAgent.view.model.SimpleModel;
import secretAgent.world.GLModel;

/**
 * Creates simple model. 
 * 
 * @author Ondrej Milenovsky
 */
public class SimpleModelCreator extends AbstractModelCreator {
    private static final long serialVersionUID = 620155121207717964L;

    private static final String FRAME_PROPERTY = "frame";

    @Override
    public SimpleModel createModel(ModelProperties modelProperties, TileId tileId, String modelName) {
        Map<String, List<TileId>> properties = modelProperties.getProperties();
        // frame
        TileId tileId2 = getSingleTileId(properties, FRAME_PROPERTY, modelName);
        if (tileId2 != null) {
            tileId = tileId2;
        }
        // scale
        double scale = 0;
        TileId scaleTile = getSingleTileId(properties, SCALE_PROPERTY, modelName);
        if (scaleTile != null) {
            scale = tileToDouble(scaleTile);
        }
        return new SimpleModel(tileId, scale);
    }

    @Override
    public ModelInfo createModelInfo(TileId tileId, GLModel model, String modelName) {
        return new ModelInfo(model);
    }

}

File: src/main/kotlin/secretAgent/io/campaign/model/SimpleModelCreator.kt
package secretAgent.io.campaign.model

import cz.wa.secretagent.worldinfo.graphics.ModelInfo
import secretAgent.io.tiles.ModelProperties
import secretAgent.view.model.SimpleModel
import secretAgent.view.renderer.TileId
import secretAgent.world.GLModel

/**
 * Creates simple model.
 *
 * @author Ondrej Milenovsky
 */
class SimpleModelCreator : AbstractModelCreator() {

    override fun createModel(modelProperties: ModelProperties, tileId: TileId, modelName: String): SimpleModel {
        val properties = modelProperties.properties
        // frame
        val tileID = getSingleTileId(properties, FRAME_PROPERTY, modelName) ?: tileId
        // scale
        val scaleTile = getSingleTileId(properties, SCALE_PROPERTY, modelName)
        val scale = scaleTile?.let(::tileToDouble) ?: 0.0

        return SimpleModel(tileID, scale)
    }

    override fun createModelInfo(tileId: TileId, model: GLModel, modelName: String) = ModelInfo(model)

    companion object {
        private const val serialVersionUID = 620155121207717964L
        private const val FRAME_PROPERTY = "frame"
    }
}